src/EventSourcing/Domain/Aggregate/Aggregate.php

```diff
@@ -0,0 +1,46 @@
+<?php
+
+declare(strict_types=1);
+
+namespace EventSourcingWorkshop\EventSourcing\Domain\Aggregate;
+
+/**
+ * An aggregate is an object representation of the state of a system, based on its past history.
+ *
+ * When interacting with an aggregate, we don't mutate its state, but rather emit events: this
+ * is not represented by the interface itself, other than the fact that all aggregates are immutable.
```

Comment:
This is a nice twist, when compared to other aggregate designs in the PHP world: a lot of confusion spawned (in the past) when interacting with mutable aggregates that:

 1. produced events
 2. applied those events internally

By making the aggregate immutable, this confusion is completely gone, and I can instead have workshop attendees follow the hints given by `vimeo/psalm` on whether they are doing something that is not valid.